contracts/BidderRegistry.sol

```diff
@@ -163,22 +163,28 @@ contract BidderRegistry is IBidderRegistry, Ownable, ReentrancyGuard {
      */
     function retrieveFunds(
         bytes32 commitmentDigest,
-        address payable provider
+        address payable provider,
+        uint256 residualBidAfterDecay
     ) external nonReentrant onlyPreConfirmationEngine {
 
         BidState memory bidState = BidPayment[commitmentDigest];
         require(bidState.state == State.PreConfirmed, "The bid was not preconfirmed");
         uint256 amt = bidState.bidAmt;
-        uint256 feeAmt = (amt * uint256(feePercent) * PRECISION) / PERCENT;
-        uint256 amtMinusFee = amt - feeAmt;
+        uint256 decayedAmt = ( bidState.bidAmt * residualBidAfterDecay * PRECISION) / PERCENT;
```

Comment:
nit: can use `amt` variable here in place of `bidState.bidAmt` or delete the var

Reply:
Thanks deleting the var